src/bin/ion/commands/beta/generate/model.rs

```diff
@@ -0,0 +1,154 @@
+use ion_schema::isl::isl_type::IslType;
+use std::collections::HashMap;
+use std::ops::Deref;
+// This module contains a data model that code generator cna use to render a template based on the type of the model.
+// Currently, this same data model represented is by `AbstractDataType` but it doesn't hold all the information for the template.
+// e.g. currently there are different fields in template that hold this information like fields, target_kind_name, abstract_data_type.
+// Also, current approach doesn't allow having nested sequences in the generated code. Because the `element_type` in `AbstractDataType::Sequence`
+// doesn't have information on its nested types' `element_type`. This can be resolved with below defined new data model.
+// _Note: This model will eventually use a map (FullQualifiedTypeReference, DataModel) to resolve some the references in container types(sequence or structure)._
+// TODO: This is not yet used in the implementation, modify current implementation to use this data model.
+use crate::commands::beta::generate::context::SequenceType;
+use serde::Serialize;
+
+#[allow(dead_code)]
+#[derive(Debug, Clone, PartialEq)]
+pub struct DataModel {
```

Comment:
I find this name a bit confusing. To me, a data model is a collection of data types. In the context of codegen, I assumed it would represent all of the types in the ISL schema.

What do you think of renaming these:
* `DataModel` -> `AbstractDataType`
* `CodeGenType` -> `AbstractDataKind`
?